### ISSUE TITLE
driver/Decadac Voltage scaling

### ISSUE THREAD

**Issue by Dominik-Vogel** [QCoDeS/Qcodes#904]
### Problems
The Decadac itself, the slots and the channels initialize with minimum and maximum values for every channel at +-5V:
```
class Decadac(VisaInstrument, DacReader):
    def __init__(self, name, address, min_val=-5, max_val=5, **kwargs):
...
class DacSlot(InstrumentChannel, DacReader):
    def __init__(self, parent, name, slot, min_val=-5, max_val=5):
...
class DacChannel(InstrumentChannel, DacReader):
    def __init__(self, parent, name, channel, min_val=-5, max_val=5):
```
Problems with this are:
 - the actual supported hardware limits are [-10,0], [-10, 10] and [0,10]. [-5, 5] is not valid
 - it is complicated to set different ranges for different channels

### Hardware limits
The `min_val` and `max_val` attributes have two purposes:
* The Decadac driver scales the supplied voltage from float to an integer according to their values
* they serve as limits for value validation
On the front panel of the decadac is a three state switch, that allows to set the ranges as described above. So any other setting than one of those does not make any sense in terms of conversion. If other ranges are wished for in validation, this should be implemented separately.
-> we should implement hardware limist in three modes -1, 0, 1 corresponding to the three ranges
-> limits for validation should be added independently (but of course compatible)

### Per channel setting
The `min_val` and `max_val` attributes can simply be set per channel to achieve the right scaling. But this leads to problems with validators and ramps that will not update
-> we should add the option to provide `min_val` and `max_val` as dictionary to the constructor and maybe supply a method to change those.

Once there is time I can make a PR. For now I just wanted to share the issue and get feedback, especially from @jenshnielsen and @WilliamHPNielsen who wrote the driver.

**Comment by WilliamHPNielsen** [comment]
Good catch. This is really important as the `min_val` and `max_val` are used in the conversion and therefore the conversion is currently WRONG! (the `_dac_v_to_code` function). Please change this ASAP.

I generally agree with your outline of a solution; have a variable to store the voltage mode (1, 0, -1), use that for conversion, and allow users to specify other compliance limits on top of that. I don't understand what you mean by 
>The min_val and max_val attributes can simply be set per channel to achieve the right scaling. But this leads to problems with validators and ramps that will not update

Validators should be handled by user-defined compliance limits implemented in software as parameters. Ramps? What's the problem here?
